ts/segment.go

```diff
@@ -40,12 +40,12 @@ type Segment struct {
 }
 
 // Len returns the length of the head and tail.
-func (s *Segment) Len() int {
+func (s Segment) Len() int {
 	return len(s.Head) + len(s.Tail)
 }
 
 // Finalize will release resources kept by the segment if pooled.
-func (s *Segment) Finalize() {
```

Comment:
Only thing that's not great about making this a non-pointer type is that now it won't actually `nil` out the `Head` and `Tail` of whoever is calling these methods no?